internal/apif/runner.go

```diff
@@ -0,0 +1,181 @@
+package apif
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/saucelabs/saucectl/internal/apitesting"
+	"github.com/saucelabs/saucectl/internal/job"
+	"github.com/saucelabs/saucectl/internal/region"
+	"github.com/saucelabs/saucectl/internal/report"
+	"github.com/saucelabs/saucectl/internal/tunnel"
+)
+
+// ApifRunner represents an executor for api tests
+type ApifRunner struct {
+	Project       Project
+	Client        apitesting.Client
+	Region        region.Region
+	Reporters     []report.Reporter
+	Async         bool
+	TunnelService tunnel.Service
+}
+
+// RunProject runs the tests defined in apif.Project
+func (r *ApifRunner) RunProject() (int, error) {
+	exitCode := 1
+	if err := tunnel.ValidateTunnel(r.TunnelService, r.Project.Sauce.Tunnel.Name, r.Project.Sauce.Tunnel.Owner, tunnel.V2AlphaFilter, false); err != nil {
+		return 1, err
+	}
+
+	passed := r.runSuites()
+	if passed {
+		exitCode = 0
+	}
+	return exitCode, nil
+}
+
+func (r *ApifRunner) runSuites() bool {
+	results := make(chan []apitesting.TestResult)
+	expected := 0
+
+	for _, s := range r.Project.Suites {
+		suite := s
+
+		if len(suite.Tags) == 0 && len(suite.Tests) == 0 {
```

Comment:
Maybe a small comment above would be nice to describe what this means (what the use case is). Based on the log line below, I guess this just runs the entire project?